packages/instantsearch.js/src/templates/carousel/carousel.tsx

```diff
@@ -0,0 +1,85 @@
+/** @jsx h */
+
+import { html } from 'htm/preact';
+import {
+  createCarouselComponent,
+  generateCarouselId,
+} from 'instantsearch-ui-components';
+import { Fragment, h } from 'preact';
+import { useRef } from 'preact/hooks';
+
+import type {
+  CarouselProps as CarouselUiProps,
+  VNode,
+} from 'instantsearch-ui-components';
+
+const Carousel = createCarouselComponent({
+  createElement: h,
+  Fragment,
+});
+
+function CarouselWithRefs<TObject extends Record<string, unknown>>(
+  props: Omit<
+    CarouselUiProps<TObject>,
+    'listRef' | 'nextButtonRef' | 'previousButtonRef' | 'carouselIdRef'
+  >
+) {
+  const carouselRefs: Pick<
+    CarouselUiProps<TObject>,
+    'listRef' | 'nextButtonRef' | 'previousButtonRef' | 'carouselIdRef'
+  > = {
+    listRef: useRef(null),
+    nextButtonRef: useRef(null),
+    previousButtonRef: useRef(null),
+    carouselIdRef: useRef(generateCarouselId()),
+  };
+
+  return <Carousel {...carouselRefs} {...props} />;
+}
+
+type Template = (params: { html: typeof html }) => VNode | VNode[] | null;
+
+type CreateCarouselTemplateProps<TObject extends Record<string, unknown>> = {
+  templates?: Partial<{
+    previous: Exclude<Template, string>;
+    next: Exclude<Template, string>;
+  }>;
+  cssClasses?: Partial<CarouselUiProps<TObject>['classNames']>;
+};
+
+type CarouselTemplateProps<TObject extends Record<string, unknown>> = Pick<
+  CarouselUiProps<TObject>,
+  'items'
+> & {
+  templates: {
+    item?: CarouselUiProps<TObject>['itemComponent'];
+  };
+};
+
+export function carousel<TObject extends Record<string, unknown>>({
+  cssClasses,
+  templates,
```

Comment:
```suggestion
  templates = {},
```

if code ~ like this, you don't need to check `templates?.` later

Reply:
Yup! Done in [77f0549](https://github.com/algolia/instantsearch/pull/6314/commits/77f0549d51380c2fc7e11659806810a107bd88e0).